docs/hazmat/primitives/asymmetric/dsa.rst

```diff
@@ -74,8 +74,55 @@ provider.
     ...     key_size=1024,
     ...     backend=default_backend()
     ... )
-    >>> signer = private_key.signer(hashes.SHA256())
     >>> data = b"this is some data I'd like to sign"
+
+Signing using a :class:`cryptography.hazmat.primitives.hashes.HashContext` (preferred method):
+
+.. doctest::
+
+    >>> digester = hashes.Hash(hashes.SHA256(), backend=default_backend())
+    >>> digester.update(data)
+    >>> signature = private_key.sign(digester)
+
+One-line signing:
+
+.. doctest::
+
+    >>> signature = private_key.sign(data, hashes.SHA256())
```

Comment:
Derp, sorry I was expanding this to RSA in my mind. You are correct that there's no padding arg for DSA (or ECDSA).